src/include/platform/internal/GenericPlatformManagerImpl.cpp

```diff
@@ -257,7 +257,7 @@ void GenericPlatformManagerImpl<ImplClass>::_DispatchEvent(const ChipDeviceEvent
 
     // TODO: make this configurable
 #if CHIP_PROGRESS_LOGGING
-    uint32_t delta = (static_cast<uint32_t>(System::Clock::GetMonotonicMicroseconds() - startUS)) / 1000;
+    uint32_t delta = (static_cast<uint32_t>(System::SystemClock().GetMonotonicMicroseconds() - startUS)) / 1000;
```

Comment:
The static cast should have the `/ 1000` within it, rather than outside. Otherwise it's possible that it will overrun after not that long.

Reply:
I believe this is outside the scope of the PR: you are  right this is a problem  so I created #10144, however should not block this PR that just renames Clock to SystemClock()